kong/plugins/zipkin/opentracing.lua

```diff
@@ -50,57 +53,67 @@ local function ip_tag(addr)
   end
 end
 
+
+-- adds the proxy span to the opentracing context, unless it already exists
+local function get_or_add_proxy_span(opentracing, timestamp)
+  if not opentracing.proxy_span then
+    local request_span = opentracing.request_span
+    opentracing.proxy_span = request_span:start_child_span(
+      request_span.name .. " (proxy)",
+      timestamp)
+    opentracing.proxy_span:set_tag("span.kind", "client")
+  end
+  return opentracing.proxy_span
+end
+
+
 if subsystem == "http" then
   function OpenTracingHandler:initialise_request(conf, ctx)
     local tracer = self:get_tracer(conf)
     local req = kong.request
     local wire_context = tracer:extract("http_headers", req.get_headers()) -- could be nil
-    local method, url
     local path_with_query = req.get_path_with_query()
-    if path_with_query ~= "" then
-      method = req.get_method()
-      url = req.get_scheme() .. "://" .. req.get_host() .. ":"
-        .. req.get_port() .. path_with_query
-    end
+    local method = req.get_method()
+    local url = req.get_scheme() .. "://" .. req.get_host() .. ":"
+             .. req.get_port() .. path_with_query
     local forwarded_ip = kong.client.get_forwarded_ip()
-    local request_span = tracer:start_span("kong.request", {
-      child_of = wire_context;
+
+    local request_span = tracer:start_span(method .. " " .. url, {
+      child_of = wire_context,
       start_timestamp = ngx.req.start_time(),
       tags = {
-        component = "kong";
-        ["span.kind"] = "server";
-        ["http.method"] = method;
-        ["http.url"] = url;
-        [ip_tag(forwarded_ip)] = forwarded_ip;
-        ["peer.port"] = kong.client.get_forwarded_port();
+        component = "kong",
+        ["span.kind"] = "server",
+        ["http.method"] = method,
+        ["http.url"] = url,
```

Comment:
Sorry, I am late for this. We usually don't record `http.url`, instead we record `http.path` because `url` can leak some private information like tokens or api-key. I can open a PR for this.

Reply:
Hi, we haven't released a new official version of the plugin yet, so there's still time for this. If you feel confident, please open a PR. If not, I will give it a try.